### PR TITLE
Add test for concatenated short options with an optional argument.

### PR DESCRIPTION
**Pull request by cxj**
I believe this test is sufficient for making this feature work, but there might be some edge cases which would warrant more tests for really complete code
coverage and regression testing.  They might include the following:
1. Test for an intentional failure of a required argument.
2. Test for success with absent optional argument.

Partial solution for issue #58.

### CODE REVIEW

**Comment by harikt** [comment]
@kenjis would you like to take a look into this before 4.x release ?

**Comment by kenjis** [comment]
@harikt Okay, I will.

**Comment by kenjis** [comment]
@cxj @harikt I think the test in this PR is okay.
But this PR includes only test case. And the test fails now in `4.x`.

```
$ composer test
> ./vendor/bin/phpunit
PHPUnit 8.5.19 by Sebastian Bergmann and contributors.

..............F.......................                            38 / 38 (100%)

Time: 115 ms, Memory: 6.00 MB

There was 1 failure:

1) Aura\Cli\Context\GetoptParserTest::testParse_shortClusterOptional
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
 Array &0 (
     '-f' => true
     '-z' => true
-    '-b' => 'foo_arg'
+    '-b' => true
+    0 => 'foo_arg'
 )

/Users/kenji/work/aura/Aura.Cli/tests/Context/GetoptParserTest.php:328

FAILURES!
Tests: 38, Assertions: 94, Failures: 1.
Script ./vendor/bin/phpunit handling the test event returned with error code 1
```

**Comment by kenjis** [comment]
The change below in https://github.com/auraphp/Aura.Cli/issues/58#issuecomment-321447672 can make all tests pass.

```diff
--- a/src/Context/GetoptParser.php
+++ b/src/Context/GetoptParser.php
@@ -409,6 +409,7 @@ class GetoptParser
             if (! $this->shortOptionRequiresValue($option, $name)) {
                 $this->setValue($option, true);
             }
+            $this->setShortFlagValue($name);
         }
     }
 
```

```
$ composer test
> ./vendor/bin/phpunit
PHPUnit 8.5.19 by Sebastian Bergmann and contributors.

......................................                            38 / 38 (100%)

Time: 117 ms, Memory: 6.00 MB

OK (38 tests, 94 assertions)
```

**Comment by harikt** [comment]
@kenjis I am gng to merge this PR as this has a failing test. 

If possible send the proposed fix as a PR.
